### PR TITLE
Assignment delivery

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class PasswordResetsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    user.send_password_reset if user
+    redirect_to root_path, :notice => "Reset password email sent!"
```
Comment (ghost) [inline app/controllers/password_resets_controller.rb]: Inconsistent. Is there a specific reason the '=>' was used? If not, maybe stick with something like "notice: ..."

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    @comment = Comment.new
+    comments = @post.comments
+    respond_to do |format|
+      format.html { render }
+      format.json { render json: @post}
+    end
+  end
+
+  def index
+    if params[:q]
```
Comment (malberton) [inline app/controllers/posts_controller.rb]: "q" is not particularly descriptive. I'm assuming this is a carryover from Awesome Answers (short for "question")?

File: app/jobs/top_ten_commented_posts_job.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                       limit(10).
+                       to_a
+    admins = User.where("admin = ?", true)
+    posts = post_ids.map  do |comment|
+      Post.find comment.post_id
+    end
+    admins.each do |admin|
+      PostsMailer.send_top_commented_posts(posts, admin.id).deliver_now
+    end
+    print "I'm in the Job!"
```
Comment (ghost) [inline app/jobs/top_ten_commented_posts_job.rb]: Take your tests out after you finish with them

File: app/mailers/posts_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class PostsMailer < ApplicationMailer
+  def send_top_commented_posts(posts, admin_id)
+    @posts = posts
+    @admin = User.find admin_id
+    mail(to: @admin.email, subject: "Top 10 Commented Posts")
+    print "I'm in the mailer!"
```
Comment (ghost) [inline app/mailers/posts_mailer.rb]: Take tests out when you finish

File: app/services/post_service.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class PostService
+
+  def initialize()
+
+  end
+
+  def send_email
+    # put logic here
```
Comment (ghost) [inline app/services/post_service.rb:8]: Is anything supposed to go in here?

Comment (kaiaolson) [inline app/services/post_service.rb:8]: No - Derek was showing me something.

**Pull request merged**